thinc/model.py

```diff
@@ -462,19 +462,40 @@ def copy(self: SelfT) -> SelfT:
         layers will also be deep-copied. The copy will receive a distinct `model.id`
         value.
         """
+        return self._copy({})
+
+    def _copy(self: SelfT, seen: Dict[int, Union["Model", Shim]]) -> SelfT:
```

Comment:
It seems like it might be nicer if there were a default provided here so you don't have to call `self._copy({})`, but we don't have `SimpleFrozenDict` in `thinc` so I'm not sure if there's a better option?

Reply:
That was what I had originally and changed it to this based on earlier feedback :-). I personally prefer this version because it avoids exposing the details of the recursion in the public interface definition.

Reply:
I just meant the API for `_copy()`, where I think having to call `.copy({})` is a bit clunky.

Reply:
Oh sorry, now I've got you — yes, that is a definite improvement.